WordPress/WordPress.xcodeproj/project.pbxproj

```diff
@@ -8084,11 +8085,11 @@
 			);
 			name = "Generate API credentials";
 			outputPaths = (
-				"$(SRCROOT)/Derived Sources/ApiCredentials.m",
+				"$(BUILT_PRODUCTS_DIR)/../Derived Sources/ApiCredentials.m",
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "DERIVED_SOURCES_DIR=\"${SOURCE_ROOT}/Derived Sources\"\ncp \"${SOURCE_ROOT}/Credentials/ApiCredentials.m\" \"${DERIVED_SOURCES_DIR}/ApiCredentials.m\"\n\necho \"Checking for WordPress.com Oauth App Secret in $WPCOM_CONFIG\"\nif [ -a $WPCOM_CONFIG ]\nthen\necho \"Config found\"\nsource $WPCOM_CONFIG\nelse\necho \"No config found\"\nexit 0\nfi\n\nif [ -z $WPCOM_APP_ID ]\nthen\necho \"warning: Missing WPCOM_APP_ID\"\nexit 1\nfi\nif [ -z $WPCOM_APP_SECRET ]\nthen\necho \"warning: Missing WPCOM_APP_SECRET\"\nexit 1\nfi\n\necho \"Generating credentials file in ${DERIVED_SOURCES_DIR}/ApiCredentials.m\"\nruby \"${SOURCE_ROOT}/Credentials/gencredentials.rb\" > \"${DERIVED_SOURCES_DIR}/ApiCredentials.m\"\n";
+			shellScript = "# Delete the in-repo 'Derived Sources'\nrm -rf \"${SOURCE_ROOT}/WordPress/Derived Sources\"\n\n# Place 'Derived Sources' relative to BUILT_PRODUCTS_DIR \nDERIVED_SOURCES_DIR=\"${BUILT_PRODUCTS_DIR}/../Derived Sources\"\n\nmkdir -p \"${DERIVED_SOURCES_DIR}\"\ncp \"${SOURCE_ROOT}/Credentials/ApiCredentials.m\" \"${DERIVED_SOURCES_DIR}/ApiCredentials.m\"\n\necho \"Checking for WordPress.com Oauth App Secret in $WPCOM_CONFIG\"\nif [ -a $WPCOM_CONFIG ]\nthen\necho \"Config found\"\nsource $WPCOM_CONFIG\nelse\necho \"No config found\"\nexit 0\nfi\n\nif [ -z $WPCOM_APP_ID ]\nthen\necho \"warning: Missing WPCOM_APP_ID\"\nexit 1\nfi\nif [ -z $WPCOM_APP_SECRET ]\nthen\necho \"warning: Missing WPCOM_APP_SECRET\"\nexit 1\nfi\n\necho \"Generating credentials file in ${DERIVED_SOURCES_DIR}/ApiCredentials.m\"\nruby \"${SOURCE_ROOT}/Credentials/gencredentials.rb\" > \"${DERIVED_SOURCES_DIR}/ApiCredentials.m\"\n";
```

Comment:
Since this is not very readable, here are the script changes in full:

```diff
+# Delete the in-repo 'Derived Sources'
+rm -rf "${SOURCE_ROOT}/WordPress/Derived Sources"
+
+# Place 'Derived Sources' relative to BUILT_PRODUCTS_DIR
-DERIVED_SOURCES_DIR=\"${SOURCE_ROOT}/Derived Sources\"
+DERIVED_SOURCES_DIR="${BUILT_PRODUCTS_DIR}/../Derived Sources"

+mkdir -p "${DERIVED_SOURCES_DIR}"
cp "${SOURCE_ROOT}/Credentials/ApiCredentials.m" "${DERIVED_SOURCES_DIR}/ApiCredentials.m"

echo "Checking for WordPress.com Oauth App Secret in $WPCOM_CONFIG"
if [ -a $WPCOM_CONFIG ]
then
echo "Config found"
source $WPCOM_CONFIG
else
echo "No config found"
exit 0
fi

if [ -z $WPCOM_APP_ID ]
then
echo "warning: Missing WPCOM_APP_ID"
exit 1
fi
if [ -z $WPCOM_APP_SECRET ]
then
echo "warning: Missing WPCOM_APP_SECRET"
exit 1
fi

echo "Generating credentials file in ${DERIVED_SOURCES_DIR}/ApiCredentials.m"
ruby "${SOURCE_ROOT}/Credentials/gencredentials.rb" > "${DERIVED_SOURCES_DIR}/ApiCredentials.m"
```

Reply:
Tested it and the script doesn't delete the old folder. May the right path for `rm -rf` be: `${SOURCE_ROOT}/Derived Sources`.